docs/ref/contrib/admin/index.txt

```diff
@@ -1812,6 +1812,16 @@ templates used by the :class:`ModelAdmin` views:
     This uses the ``HttpRequest`` instance to filter the ``Car`` foreign key
     field to only display the cars owned by the ``User`` instance.
 
+    A ``ModelForm`` can be used to filter with ``instance``, as seen in
+    :ref:`fields-which-handle-relationships`.
+
+    For example, by declaring :attr:`ModelAdmin.form`, you will be able to use a custom queryset::
+
+        class MyCarAdminForm(forms.ModelForm):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                self.fields["garage"].queryset = self.instance.owner.garage_set.all()
```

Comment:
I'm still not convinced that it's the best example, we should at least add a guard for an empty instance:
```python
if self.instance.pk:
    self.fields['garage'].queryset = ...
```
We could also use an example from the first version #12101 of this patch and get rid of it:
```
    ... For example::

        class CountryAdminForm(forms.ModelForm):
            def __init__(self, *args, **kwargs):
                super().__init__(*args, **kwargs)
                self.fields['capital'].queryset = City.objects.filter(country=self.instance.pk)
                    
        class CountryAdmin(admin.ModelAdmin):
            form = CountryAdminForm
```

Reply:
Updated
<img width="658" alt="Screen Shot 2020-07-13 at 09 31 58" src="https://user-images.githubusercontent.com/55533/87304836-d2ba3a00-c4eb-11ea-8742-5b16da834516.png">